fta/samples/api/views.py

```diff
@@ -47,48 +47,45 @@ def add_sample(self, request, format=None):
         return Response({"id": sample.id})
 
 
-class AddFathomSampleViewSet(viewsets.ViewSet):
-    basename = "add_fathom_sample"
+class AddLabeledSampleViewSet(viewsets.ViewSet):
+    basename = "add_labeled_sample"
 
     @action(detail=False, methods=["post"])
-    def add_fathom_sample(self, request, format=None):
-        frozen_page = request.data["frozen_page"]
+    def add_labeled_sample(self, request, format=None):
+        labeled_page = request.data["labeled_page"]
         freeze_software = request.data["freeze_software"]
         notes = request.data["notes"] if "notes" in request.data else ""
-        sample = sample_from_required(frozen_page, freeze_software, notes)
-        existing_sample = Sample.objects.filter(url__contains=sample.url).order_by(
-            "-id"
-        )[:1]
+        labeled_sample_id = request.data["labeled_sample_id"]
+
+        # 1. Get the LabeledSample
+        try:
+            existing_labeled_sample = LabeledSample.objects.get(id=labeled_sample_id)
+        except LabeledSample.DoesNotExist:
+            existing_labeled_sample = None
 
-        # If the sample already exists reuse it.
-        if existing_sample.exists() is False:
+        if not existing_labeled_sample:
+            # 2. If the LabeledSample does not exist create  first create the Sample
+            sample = sample_from_required(labeled_page, freeze_software, notes)
             sample.save()
         else:
-            sample = existing_sample.first()
-
-        # check if a LabeledSample already exists for the sample.
-        existing_labeled_sample = LabeledSample.objects.filter(
-            original_sample_id=sample.id
-        )
-
-        id_of_existing_labeled_sample = -1
-        if existing_labeled_sample.exists() is True:
-            id_of_existing_labeled_sample = existing_labeled_sample.first().id
+            # 2. Get the associated original sample.  Should always exist due to FK and cannot be Null
+            sample = Sample.objects.get(id=existing_labeled_sample.original_sample_id)
```

Comment:
This should be able to be simpler:
```suggestion
            sample = existing_labeled_sample.original_sample
```